### PR TITLE
{Jmana} add website support

### LINKED ISSUE #2637: [Site Request] Jmana1
**Name of the website**
Jmana1

**Website urls (examples below)**
- Site: https://jmana1.net/
- Manga List: https://jmana1.net/comic_main
- Manga example: https://jmana1.net/book?bookname=%EC%A0%84%EC%83%9D%ED%96%88%EB%8D%94%EB%8B%88+%EC%8A%AC%EB%9D%BC%EC%9E%84%EC%9D%B4%EC%97%88%EB%8D%98+%EA%B1%B4%EC%97%90+%EB%8C%80%ED%95%98%EC%97%AC
- Chapter online viewer example: https://jmana1.net/bookdetail?bookname=%EC%A0%84%EC%83%9D%ED%96%88%EB%8D%94%EB%8B%88+%EC%8A%AC%EB%9D%BC%EC%9E%84%EC%9D%B4%EC%97%88%EB%8D%98+%EA%B1%B4%EC%97%90+%EB%8C%80%ED%95%98%EC%97%AC&bookdetail=403007

**Languages**
Korean

### PR DESCRIPTION
**Pull request by manh21**
Resolve #2637

### CODE REVIEW

File: src/web/mjs/connectors/Jmana1.mjs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        return data.map(el => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(el, this.url),
+                title: el.textContent.replace(manga.title, '').trim(),
+            };
+        });
+    }
+
+    // TODO: address 403 Forbidden
+    async _getPages(chapter) {
```
Comment (manh21) [inline src/web/mjs/connectors/Jmana1.mjs:57]: can't load image URL [https://img3.xyz.futbol/comics/jdrive02/202011/others/89c11a05-4d72-4727-936d-d64748ec96e3.jpg](https://img3.xyz.futbol/comics/jdrive02/202011/others/89c11a05-4d72-4727-936d-d64748ec96e3.jpg) 403 Response

Comment (27136773) [reply src/web/mjs/connectors/Jmana1.mjs:57]: Try [add referer header](https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/Hitomi.mjs#L12) and [return a special connector URL](https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/Hitomi.mjs#L51) like in Hitomi or MangaDex

image

File: src/web/mjs/connectors/Jmana1.mjs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        for(let page = 1, run = true; run; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangas.length ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        // NOTE: Avoid 429 Too Many Request
+        await new Promise(r => setTimeout(r, 2000));
```
Comment (27136773) [inline src/web/mjs/connectors/Jmana1.mjs]: May use `await this.wait(2000)`

File: src/web/mjs/connectors/Jmana1.mjs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        const uri = new URL(this.path, this.url);
+        uri.searchParams.set('page', page);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.querySelector('span.titBox span.price').textContent.trim()
```
Comment (27136773) [inline src/web/mjs/connectors/Jmana1.mjs:40]: The nam e os the CSS class `.price` is a little bit suspicious.
Does this work with free, non-purchased and also purchased content?

File: src/web/mjs/connectors/Jmana1.mjs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryPages);
+        return data.map(element => {
+            return this.getAbsolutePath(element.dataset['src'] || element['srcset'] || element, request.url);
+        });
+    }
+
+    async _getMangaFromURI(uri) {
+        const urlParams = new URLSearchParams(uri.search);
+        const bookname = urlParams.get('bookname');
```
Comment (27136773) [inline src/web/mjs/connectors/Jmana1.mjs]: May use `uri.searchParams.get('bookname');` directly

File: src/web/mjs/connectors/Jmana1.mjs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        const id = `/book?bookname=${bookname}`;
+        const title = bookname.replace('+', ' ');
```
Comment (27136773) [inline src/web/mjs/connectors/Jmana1.mjs]: Is it possible to parse the title from HTML tags?
I have doubts that the parameter always exactly matches the title that is shown in the manga list (haven't found proof for this yet), e.g.
`h1.section-title` => textContent

File: src/web/mjs/connectors/Jmana1.mjs (hunk 3 above)
Comment (manh21) [reply src/web/mjs/connectors/Jmana1.mjs:40]: This is just how the site structure works for all titles

**Review by 27136773 — APPROVED** [review]
Looks good

**Pull request merged**